server/sl/user_messages.go

```diff
@@ -14,7 +14,7 @@ func (sl *sl) dmUserWelcomeToSolarLottery(user *User) {
 	// There is the special case when a user uses the plugin for the first time,
 	// in which case the actingUser is not yet set. Default to the "subject" user.
 	actingUser := sl.actingUser
-	if actingUser == nil {
+	if actingUser == nil || actingUser.mattermostUser == nil {
```

Comment:
I am a little confused how there may be an unexpanded `sl.actingUser` here, and arguably if there was, perhaps a more "idiomatic" thing to do would be to use it, and expand.

Reply:
actingUser here doesn't even has MattermostUserID, so expanding is impossible

Reply:
`actingUser` should always have `MattermostUserID`, could we investigate when and why it is happening?

Reply:
So, if first ever command is `/lotto user join/qualify/leave/disqualify rotation` and `withExpandedActingUser` is called in `sl_setup.go`, it first calls `sl.loadOrMakeUser` and then sets returned user as `sl.actingUser`(before `sl.actingUser` is just newUser(""), so without `MattermostUserID`), `sl.loadOrMakeUser` itself, calls `storeUserWelcomeNew` and as User is new `sl.dmUserWelcomeToSolarLottery` is called, where `sl.actingUser` doesn't have `MattermostUserID` as it hasn't been set yet. 
Solution to this may be to remove first line of `withExpandedActingUser` in `sl_setup.go` which creates `sl.actingUser` with no parameters, as `sl.actingUser` should be null in `dmUserWelcomeToSolarLottery` and `actingUser` should become `user` after this check

Reply:
There's definitely a catch-22 there somewhere, I agree, I'm ok with committing as is, and will try to streamline it later.